### ISSUE TITLE
"block" keyword unrecognized in 2.1 in notebook backend

### ISSUE THREAD

**Issue by juliotux** [matplotlib/matplotlib#9651]
Aparently, when I call `plt.show(block=True)` in a notebook, using notebook backend set by `%matplotlib notebook`, the `block` keyword is unrecognized.

```
from matplotlib import pyplot as plt
%matplotlib notebook

f, ax = plt.subplots(1, 1)
ax.plot([1,2,3], [3,2,1], 'k-')
plt.show(block=True)
```
results in:
```
~/.anaconda3/lib/python3.6/site-packages/matplotlib/pyplot.py in show(*args, **kw)
    249     """
    250     global _show
--> 251     return _show(*args, **kw)
    252 
    253 

TypeError: show() got an unexpected keyword argument 'block'
```

My system:
Antergos Linux
Matplotlib: 2.1.0
IPython :  6.1.0
Notebook: 5.0.0
Python: 3.6 (anaconda)

**Comment by juliotux** [comment]
Complementing:

The same occurs with:
```
from matplotlib import pyplot as plt
%matplotlib notebook

f, ax = plt.subplots(1, 1)
ax.plot([1,2,3], [3,2,1], 'k-')
f.show(block=True)
```

**Comment by tacaswell** [comment]
I suspect this has always been the case and it did not get implemented because what blocking would mean in the context of notebook.  The correct thing is probably to let nbagg's `show` take the kwarg and ignore it.

**Comment by dsblank** [comment]
Fixed in 4516ebf28ac

**Comment by tacaswell** [comment]
Closed by https://github.com/matplotlib/matplotlib/pull/9733

Thanks again @dsblank !

**Comment by jklymak** [comment]
Closed by #9733 Thanks @dsblank and @juliotux
